File: benchmarks/reference/c/call_stress.c
/* call_stress.c — Function call overhead benchmark (100K iterations).
   Equivalent to examples/il/benchmarks/call_stress.il */
#include <stdint.h>
#include <stdlib.h>

static int64_t add_triple(int64_t a, int64_t b, int64_t c)
{
    return a + b + c;
}

static int64_t mul_pair(int64_t x, int64_t y)
{
    return x * y;
}

static int64_t compute(int64_t n)
{
    int64_t a = n;
    int64_t b = n + 1;
    int64_t c = n + 2;
    int64_t sum = add_triple(a, b, c);
    return mul_pair(sum, 3);
}

int main(void)
{
    int64_t sum = 0;
    for (int64_t i = 0; i < 100000; ++i) {
        int64_t r1 = compute(i);
        int64_t r2 = add_triple(i, r1, 1);
        int64_t r3 = mul_pair(r2, 2);
        sum += r3;
    }
    return (int)(sum & 0xFF);
}

File: benchmarks/reference/csharp/CallStress.cs
/* call_stress — Function call overhead benchmark (10M iterations).
   Equivalent to examples/il/benchmarks/call_stress.il */
using System;
using System.Runtime.CompilerServices;

class CallStress
{
    [MethodImpl(MethodImplOptions.NoInlining)]
    static long AddTriple(long a, long b, long c)
    {
        return a + b + c;
    }

    [MethodImpl(MethodImplOptions.NoInlining)]
    static long MulPair(long x, long y)
    {
        return x * y;
    }

    [MethodImpl(MethodImplOptions.NoInlining)]
    static long Compute(long n)
    {
        long a = n;
        long b = n + 1;
        long c = n + 2;
        long sum = AddTriple(a, b, c);
        return MulPair(sum, 3);
    }

    [MethodImpl(MethodImplOptions.NoInlining)]
    static long Run(long n)
    {
        long sum = 0;
        for (long i = 0; i < n; ++i)
        {
            long r1 = Compute(i);
            long r2 = AddTriple(i, r1, 1);
            long r3 = MulPair(r2, 2);
            sum += r3;
        }
        return sum;
    }

    static int Main(string[] args)
    {
        long result = Run(10000000 + args.Length);
        return (int)(result & 0xFF);
    }
}
